docs/guides/writing-a-parser.md

```diff
@@ -0,0 +1,108 @@
+# Writing a parser
+
+This guide shows how to implement a simple parser using the [alfa-parser](https://github.com/Siteimprove/alfa/tree/main/packages/alfa-parser) package. The guide is based on the work done in [#1339][PR].
+
+## Background
+
+The `alfa-parser` package contains functionality to make it easy to implement a parser. This package is used internally by Alfa to parse HTML, CSS etc., but it can be used for any parsing scenario.
+
+For this guide we want to create a parser that parses the content of the HTML attribute `autocomplete`. The content of the attribute must follow a set of rules as defined in the [HTML specification](https://html.spec.whatwg.org/multipage/#autofill-detail-tokens).
+
+## Defining the grammar
+
+We refer to the rules defined in the specification as the grammar or the syntax of the `autocomplete` attribute. It must contain a space separated list of tokens in a specific order, some of which may be optional and some may be required depending on the precense of some preceding token. For example the following is an invalid use of the attribute
+```HTML
+<input autocomplete="section-blue billing home" />
+```
+This is invalid because the `home` token must be followed by a required token.
+
+We can express the grammar as follows
+```
+[section] [addressType] (unmodifiable | [modifier] modifiable) [webauthn] EOF
+```
+where a token name sorounded by square brackets `[]` means it is an optional token, tokens seperated by a vertical line `|` means exactly one of the tokens must be present and `EOF` is a special token signaling that no more tokens are allowed.
+
+Here the possible values for the tokens are not listed, but as an example `addressType` is defined as follows
+```
+addressType = "shipping" | "billing"
```

Comment:
```suggestion
addressType ::= "shipping" | "billing"
```
Keeping it closer to [BNF](https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form) syntax